### PR TITLE
Consensus-related unit tests and bug fixes

### PR DESCRIPTION
**Pull request by AlexandraRoatis**
## Description

- Introduces the ability to load the state of the blockchain at specific blocks to check for interesting use cases.
- Bug fixes and unit tests to ensure correct sync with mastery.

Fixes Issue #78.

## Type of change

Insert **x** into the following checkboxes to confirm (eg. [x]):
- [x] Bug fix.
- [ ] New feature.
- [ ] Enhancement.
- [x] Unit test.
- [ ] Breaking change (a fix or feature that causes existing functionality to not work as expected).
- [ ] Requires documentation update.

## Testing

Please describe the tests you used to validate this pull request. Provide any relevant details for test configurations as well as any instructions to reproduce these results.

- existing unit tests
- consensus specific tests from instances encountered on the mastery network

## Verification

Insert **x** into the following checkboxes to confirm (eg. [x]):
- [x] I have self-reviewed my own code and conformed to the style guidelines of this project.
- [x] New and existing tests pass locally with my changes.
- [x] I have added tests for my fix or feature.
- [ ] I have made appropriate changes to the corresponding documentation.
- [x] My code generates no new warnings.
- [ ] Any dependent changes have been made.

### CODE REVIEW

File: modPrecompiled/src/org/aion/precompiled/PrecompiledSideEffects.java [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+     *
+     * @param log The log to add to the execution logs.
+     */
+    @Override
+    public void addLog(IExecutionLog log) {
+        logs.add(log);
+    }
+
+    /**
+     * Adds a collection of logs, logs, to the execution logs.
```
Comment (AionJayT) [inline modPrecompiled/src/org/aion/precompiled/PrecompiledSideEffects.java]: what's the second `logs` for?

Comment (arajasek) [reply modPrecompiled/src/org/aion/precompiled/PrecompiledSideEffects.java]: It's meant to be read as "This function adds a collection of logs, received as the parameter `logs`, to the execution logs". This is the general pattern of the javadoc comments, copied from `SideEffects.java` in the FVM.

Comment (arajasek) [reply modPrecompiled/src/org/aion/precompiled/PrecompiledSideEffects.java]: (I agree that it's confusing / clumsy, though)

Comment (AionJayT) [reply modPrecompiled/src/org/aion/precompiled/PrecompiledSideEffects.java]: I see.

**Pull request merged**